internal/plugin6/grpc_provider.go

```diff
@@ -748,6 +788,103 @@ func (p *GRPCProvider) ReadDataSource(ctx context.Context, r providers.ReadDataS
 	return resp
 }
 
+func (p *GRPCProvider) OpenEphemeralResource(ctx context.Context, r providers.OpenEphemeralResourceRequest) (resp providers.OpenEphemeralResourceResponse) {
+	logger.Trace("GRPCProvider.v6: OpenEphemeralResource")
+
+	schema := p.GetProviderSchema(ctx)
+	if schema.Diagnostics.HasErrors() {
+		resp.Diagnostics = schema.Diagnostics
+		return resp
+	}
+
+	ephemeralResourceSchema, ok := schema.EphemeralResources[r.TypeName]
+	if !ok {
+		schema.Diagnostics = schema.Diagnostics.Append(fmt.Errorf("unknown ephemeral resource %q", r.TypeName))
+		resp.Diagnostics = schema.Diagnostics
+		return resp
+	}
+
+	config, err := msgpack.Marshal(r.Config, ephemeralResourceSchema.Block.ImpliedType())
+	if err != nil {
+		resp.Diagnostics = resp.Diagnostics.Append(err)
+		return resp
+	}
+
+	protoReq := &proto6.OpenEphemeralResource_Request{
+		TypeName: r.TypeName,
+		Config: &proto6.DynamicValue{
+			Msgpack: config,
+		},
+	}
+
+	protoResp, err := p.client.OpenEphemeralResource(ctx, protoReq)
+	if err != nil {
+		resp.Diagnostics = resp.Diagnostics.Append(grpcErr(err))
+		return resp
+	}
+	resp.Diagnostics = resp.Diagnostics.Append(convert.ProtoToDiagnostics(protoResp.Diagnostics))
+
+	result, err := decodeDynamicValue(protoResp.Result, ephemeralResourceSchema.Block.ImpliedType())
+	if err != nil {
+		resp.Diagnostics = resp.Diagnostics.Append(err)
+		return resp
+	}
+	resp.Result = result
+	resp.Private = protoResp.Private
+	if protoResp.Deferred != nil {
+		resp.Deferred = &providers.Deferred{
+			Reason: providers.DeferredReason(protoResp.Deferred.Reason),
+		}
+	}
+	if protoResp.RenewAt != nil {
+		renewAt := protoResp.RenewAt.AsTime()
+		resp.RenewAt = &renewAt
+	}
+
+	return resp
+}
+
+func (p *GRPCProvider) RenewEphemeralResource(ctx context.Context, r providers.RenewEphemeralResourceRequest) (resp providers.RenewEphemeralResourceResponse) {
+	logger.Trace("GRPCProvider.v6: RenewEphemeralResource")
```

Comment:
Should we bother checking the schema for the ephemeral resource type here and in Close? The provider should throw an error with bad input, but maybe we should be consistent?

Reply:
Sounds well reasoned to me :+1: